proposals/0377-parameter-ownership-modifiers.md

```diff
@@ -189,26 +208,111 @@ extension String {
 let helloWorld = "hello ".plus("cruel ").plus("world")
 ```
 
+`borrowing` and `consuming` parameter values are also **not implicitly copyable**
+inside of the function or closure body:
+
+```
+func foo(x: borrowing String) -> (String, String) {
+    return (x, x) // ERROR: needs to copy `x`
+}
+func bar(x: consuming String) -> (String, String) {
+    return (x, x) // ERROR: needs to copy `x`
+}
+```
+
+And so is the `self` parameter within a method that has the method-level
+`borrowing` or `consuming` modifier:
+
+```
+extension String {
+    borrowing func foo() -> (String, String) {
+        return (self, self) // ERROR: needs to copy `self`
+    }
+    consuming func bar() -> (String, String) {
+        return (self, self) // ERROR: needs to copy `x`
+    }
+}
+```
+
+
+A value would need to be implicitly copied if:
+
+- a *consuming operation* is applied to a `borrowing` binding, or
+- a *consuming operation* is applied to a `consuming` binding after it has
+  already been consumed, or while a *borrowing* or *mutating operation* is simultaneously
+  being performed on the same binding
+
+where *consuming*, *borrowing*, and *mutating operations* are as described for
+values of noncopyable type in
+[SE-0390](https://github.com/apple/swift-evolution/blob/main/proposals/0390-noncopyable-structs-and-enums.md#using-noncopyable-values).
+In essence, disabling implicit copying for a value makes it behave as if it
+were a value of noncopyable type.
```

Comment:
I think this sentence should distinguish between the bindings and the values for clarity.